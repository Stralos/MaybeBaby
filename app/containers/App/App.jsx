// @flow
import React, { PureComponent, Fragment } from 'react';
import styled, { injectGlobal } from 'styled-components';
import { Route, NavLink, Switch } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import ROUTES from 'utils/routes';
import Nav from 'components/Nav';
import SocialMediaIcon from 'components/SocialMediaIcon';
import Services from 'components/Services';
import Home from 'containers/Home';

export type Props = {
  +venueName: string,
  +socialMedia: {
    +facebook: string,
    +instagram: string,
    +twitter: string,
  },
}

const AppStyled = styled.div`
  min-height: 100%;
  margin: 0 auto;
  max-width: 980px;
  width: 100%;
  display: flex;
  flex-direction: column;
`;

const activeClassName = 'active';
const StyledNavLink = styled(NavLink).attrs({
  activeClassName,
})`
  margin: 0 4px;
  padding: 4px 8px;
  color: #eee;
  font-size: 16px;
  text-decoration: none;
  font-weight: 400;
  letter-spacing: .1em;
  text-transform: uppercase;

  &.${activeClassName} {
    //font-weight: 600;
  }
  &:hover {
    color: #c8c8c8;
  }
`;

const Header = styled.h1`
  margin: 32px 0 24px 0;
  display: flex;
  justify-content: space-between;
  color: white;
  font-family: 'Marcellus SC';
`;

const VenueName = styled(NavLink)`
  font-family: 'Marcellus SC';
  font-size: 40px;
  line-height: 40px;
  color: #eee;
  letter-spacing: 0.05em;
  font-weight: 500;
  text-decoration: none;
  &:hover {
    color: #c8c8c8;
  }
`;

const SocialMedia = styled.div`
  font-size: 21px;
  margin: 0 -12px;
  > * {
    margin: 0 12px;
  }
`;
const RouteContent = styled.div`
  flex: 1;
`;
const Footer = styled.footer`
  background: #222;
  padding: 32px 0;
`;

class App extends PureComponent<Props> {
  componentWillMount() {
    /*
    ** the fontfamaly should be already passed inside the theme object
    ** all the calculations should be done beforehand
    ** in the theme layer.
    ** component should only display what he is given.
    */
    const globalStyles = `
      #root, body, html {
        margin: 0;
        height: 100%;
        min-height: 100%;
      }
      body, html {
        background-color: #333; 
        color: #ccc;
        font-family: 'Gentium Basic';
      }
    `;
    // eslint-disable-next-line no-unused-expressions
    injectGlobal`${globalStyles}`;
  }
  injectHtmlHead = () => {
    const noIndexTag = () => {
      return (
        <Helmet>
          <meta name="robots" content="noindex, nofollow" />
        </Helmet>
      );
    };
    return (
      <Fragment>
        <Helmet>
          <title> Test </title>
        </Helmet>
        <Switch>
          <Route exact path={ROUTES.HOME} />
          <Route render={noIndexTag} />
        </Switch>
      </Fragment>
    );
  }

  render() {
    const {
      venueName,
      socialMedia: {
        facebook, twitter, instagram,
      },
    } = this.props;
    return (
      <AppStyled>
        {this.injectHtmlHead()}
        <Header>
          <VenueName to={ROUTES.HOME}> {venueName} </VenueName>
          <SocialMedia>
            <SocialMediaIcon href={facebook}> F </SocialMediaIcon>
            <SocialMediaIcon href={twitter}> T</SocialMediaIcon>
            <SocialMediaIcon href={instagram}> I</SocialMediaIcon>
          </SocialMedia>
        </Header>
        <Nav>
          <StyledNavLink exact to={ROUTES.HOME}> Home </StyledNavLink>
          <StyledNavLink exact to={ROUTES.SERVICES}> Service & Prices </StyledNavLink>
          <StyledNavLink exact to={ROUTES.TEAM}> Our Team </StyledNavLink>
          <StyledNavLink exact to={ROUTES.GALLERY}> Gallery </StyledNavLink>
          <StyledNavLink exact to={ROUTES.CONTACT}> Contact </StyledNavLink>
        </Nav>
        <RouteContent>
          <Switch>
            <Route exact path={ROUTES.HOME} component={Home} />
            <Route exact path={ROUTES.SERVICES} component={Services} />
          </Switch>
        </RouteContent>
        <Footer>
          Footer
        </Footer>
      </AppStyled>
    );
  }
}

export default App;
