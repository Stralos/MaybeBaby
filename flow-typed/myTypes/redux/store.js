// @flow
import type { Store as ReduxStore } from 'redux';

declare type VenueImages = {
  images: Array<{
    src: string,
  }>
}

declare type EmployeeState = {
  employees: Array<Employee>,
}

declare type State = {
  +phone: string,
  +address: string,
  +email: string,
  +venueName: string,
  +image: string,
  +socialMedia: {
    +facebook: string,
    +instagram: string,
    +twitter: string,
  },
  +description: string,
  +products: Array<{
    +image: string,
  }>,
  +workingHours: Array<{
    +day: string,
    +open: boolean,
    +opensAt: string,
    +closesAt: string,
  }>,
  +services: Array<{
    +name: string,
    +products: Array<{
      +productName: string,
    }>
  }>,
} & EmployeeState & VenueImages

export type Store = ReduxStore<State>;
