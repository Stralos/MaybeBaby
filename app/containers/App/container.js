// @flow
import { connect } from 'react-redux';
import type { Connector } from 'react-redux';
import { withRouter } from 'react-router-dom';
import App from './App';
import type { Props } from './App';

const mapSateToProps = state => Object.assign({}, {
  greeting: state.greeting,
  date: state.date,
  socialMedia: {
    facebook: state.socialMedia.facebook,
    instagram: state.socialMedia.instagram,
    twitter: state.socialMedia.twitter,
  },
});

const connector : Connector<{}, Props> = connect(mapSateToProps, {});

export default withRouter(connector(App));
