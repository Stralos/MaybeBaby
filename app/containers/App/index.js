import { hot } from 'react-hot-loader';
import App from './container';

export default hot(module)(App);
