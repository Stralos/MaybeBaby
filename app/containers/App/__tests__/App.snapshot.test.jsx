/* @flow */
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import App from '../index';
import data from '../../../placeholderStore';

configure({ adapter: new Adapter() });

const props = data;
const theme = {
  color: 'white',
};

describe('App Component Snapshot-- ', () => {
  it('renders correctly', () => {
    const store = createStore(state => state, props);
    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <App />
          </ThemeProvider>
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
