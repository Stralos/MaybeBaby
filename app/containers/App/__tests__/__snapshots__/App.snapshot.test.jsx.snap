// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot should renders correctly home page 1`] = `
.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 -4px;
  border-top: 1px solid #888;
  border-bottom: 1px solid #888;
  padding: 8px 0;
}

.c5 {
  text-decoration: none;
  height: 24px;
  width: 24px;
  display: inline-block;
  background-image: url(icon-facebook.svg);
  color: white;
}

.c6 {
  text-decoration: none;
  height: 24px;
  width: 24px;
  display: inline-block;
  background-image: url(icon-twitter.svg);
  color: white;
}

.c7 {
  text-decoration: none;
  height: 24px;
  width: 24px;
  display: inline-block;
  background-image: url(icon-instagram.svg);
  color: white;
}

.c1 {
  margin: 0 auto;
  max-width: 980px;
  width: 100%;
}

.c0 {
  min-height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c9 {
  margin: 0 4px;
  padding: 4px 8px;
  color: #eee;
  font-size: 16px;
  text-decoration: none;
  font-weight: 400;
  -webkit-letter-spacing: .1em;
  -moz-letter-spacing: .1em;
  -ms-letter-spacing: .1em;
  letter-spacing: .1em;
  text-transform: uppercase;
}

.c9:hover {
  color: #c8c8c8;
}

.c2 {
  margin: 32px 0 24px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  color: white;
  font-family: 'Marcellus SC';
}

.c3 {
  font-family: 'Marcellus SC';
  font-size: 40px;
  line-height: 40px;
  color: #eee;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  font-weight: 500;
  text-decoration: none;
}

.c3:hover {
  color: #c8c8c8;
}

.c4 {
  font-size: 21px;
  margin: 0 -12px;
}

.c4 > * {
  margin: 0 12px;
}

.c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c11 {
  margin-top: 40px;
  background: #222;
  padding: 32px 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      <a
        aria-current="true"
        className="c3 active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
         
        Hair & Beauty
         
      </a>
      <div
        className="c4"
      >
        <a
          className="c5"
          href="www.facebook.com/hair&beauty"
          target="_blank"
          type="Facebook"
        />
        <a
          className="c6"
          href="www.twitter.com/hair&beauty"
          target="_blank"
          type="Twitter"
        />
        <a
          className="c7"
          href="www.instagram.com/hair&beauty"
          target="_blank"
          type="Instagram"
        />
      </div>
    </h1>
    <div
      className="c8"
    >
      <a
        aria-current="true"
        className="c9 active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
         Home 
      </a>
      <a
        aria-current={false}
        className="c9"
        href="/services"
        onClick={[Function]}
        style={undefined}
      >
         Service & Prices 
      </a>
      <a
        aria-current={false}
        className="c9"
        href="/team"
        onClick={[Function]}
        style={undefined}
      >
         Our Team 
      </a>
      <a
        aria-current={false}
        className="c9"
        href="/gallery"
        onClick={[Function]}
        style={undefined}
      >
         Gallery 
      </a>
      <a
        aria-current={false}
        className="c9"
        href="/#contact"
        onClick={[Function]}
        style={undefined}
      >
         Contact 
      </a>
    </div>
    <div
      className="c10"
    >
      &lt;Home /&gt;
    </div>
  </div>
  <footer
    className="c11"
  >
    Footer
  </footer>
</div>
`;
